docs/api/drivers/ResetReason.md

```diff
@@ -0,0 +1,39 @@
+## ResetReason
+
+Use the ResetReason interface to access hardware reset reason registers in a portable fashion.
+
+### ResetReason class reference
+
+[![View code](https://www.mbed.com/embed/?type=library)](http://os-doc-builder.test.mbed.com/docs/v5.8/mbed-os-api-doxy/classmbed_1_1_resetreason.html)
+
+### ResetReason example
+
+```c++
+#include "mbed.h"
+#include "ResetReason.h"
+
+#include <string>
+
+std::string reset_reason_to_string(const reset_reason_t reason)
+{
+  switch (reason)
+  {
+  case RESET_REASON_POWER_ON:
```

Comment:
the code should use our style, cases should be aligned to right (4 spaces)

Reply:
Fixed.